pkg/server/registry/apigroups/acorn/projects/storage.go

```diff
@@ -77,3 +84,40 @@ func (pr *projectCreater) Create(ctx context.Context, object types.Object) (type
 
 	return pr.creater.Create(ctx, object)
 }
+
+func migrateLegacyNamespaces(ctx context.Context, c kclient.Client) error {
+	// Once a project namespace is managed by the project instance, then this label will be set on it.
+	// Therefore, using this label selector ensures the migration only happens once for a namespace.
+	notManaged, err := klabels.NewRequirement(labels.AcornManaged, selection.DoesNotExist, nil)
+	if err != nil {
+		return err
+	}
+
+	nsSelector := klabels.SelectorFromSet(map[string]string{
+		labels.AcornProject: "true",
+	}).Add(*notManaged)
```

Comment:
nit: can these be built once and memoized? or are they not threadsafe?

Reply:
They are built at startup and then not needed again.

Reply:
wow, yup, carry on.